sympy/printing/tests/test_mathml.py

```diff
@@ -925,6 +925,25 @@ def test_toprettyxml_hooking():
     assert prettyxml_old2 == doc2.toprettyxml()
 
 
+def test_print_domains():
+    from sympy import Complexes, Integers, Naturals, Naturals0, Reals
+
+    assert mpp.doprint(Complexes) == '<mi mathvariant="normal">&#x2102;</mi>'
+    assert mpp.doprint(Integers) == '<mi mathvariant="normal">&#x2124;</mi>'
+    assert mpp.doprint(Naturals) == '<mi mathvariant="normal">&#x2115;</mi>'
+    assert mpp.doprint(Naturals0) == '<msub><mi mathvariant="normal">&#x2115;</mi><mn>0</mn></msub>'
+    assert mpp.doprint(Reals) == '<mi mathvariant="normal">&#x211D;</mi>'
+
+
+def test_print_AssocOp():
+    from sympy.core.operations import AssocOp
+    class TestAssocOp(AssocOp):
+        identity = 0
+
+    expr = TestAssocOp(1, 2)
+    mpp.doprint(expr) == '<mrow><mi>testassocop</mi><mn>2</mn><mn>1</mn></mrow>'
```

Comment:
This just renders like `testassocop12` AFAICT. Is that intended?

Reply:
Yes and no. It now uses the generic `AssocOp` fallback, which admittedly can be improved, but that is how the old code has selected to render that. However, in the current master it gives `AttributeError: 'Element' object has no attribute 'append'` so the main purpose is just to test that there isn't an exception... (It might have been clearer to use the `And` example above, but the idea is that it eventually should render at least `And` and `Or` in a better way so seemed better to create a new minimal `AssocOp` to test the fallback.)

Reply:
Probably some bracket would be better in the default fallback? `testassocop(1, 2)` type of rendering?

Reply:
Not sure why there is a specific `AssocOp` fallback since the `Basic` fallback would have given the bracket version anyway.